### DISCUSSION TITLE
First time attempt, issues on ignored args, odd colors

### DISCUSSION THREAD

**Discussion by fxfactorial** [d12frosted/homebrew-emacs-plus#634] [General]
Trying this out over emacsforosx, installation successful but several issues (really appreciate this effort) 

installed with 

```
brew install emacs-plus@30 --with-dbus --with-imagemagick --with-native-comp  --with-poll
```

on macOS `14.1.2 (23B92)`

1. arguments are ignored. I see that emacs is really a shell wrapper for 

```
#!/bin/bash
exec /usr/local/Cellar/emacs-plus@30/30.0.50/Emacs.app/Contents/MacOS/Emacs "$@"
```

but whenever I try opening file with emacs foo.go, it ignores the file. Why using bash anyway when default shell in macOS now is zsh, not sure if that's relevant. 
2. solarized-dark not working anymore, or wacky different colors. I have load-theme in my `~/.emacs.d/init.el` via
```
(when window-system
  (add-hook 'after-init-hook
    (lambda ()
      (global-hl-line-mode 1)
      (solaire-mode)
      (load-theme 'solarized-dark-high-contrast t))))
```

but now the colors look wacky off ((list-display..) gives back 550ish colors). I have this come up in messages too I think because of solarized-dark `face-spec-set-2: Invalid face box: :line-width, 2, :color, "#002732", :style, unspecified` not sure why 

3. Seems like my init is partially ignored? Might be related to 2 insofar as the full script init.el isn't evaled but I have code that injects into scratch buffer a text and now it doesn't happen. 
4. Not showing up in spotlight search, aka CMD+Space. Yes I did the symlink to /Applications and to /Volumes/System.../Applications but still doesn't come up.

**Comment by d12frosted** [comment]
### Argument is ignored

> whenever I try opening file with emacs foo.go

Are you sure? Just tested with and without my personal configurations and it worked:

```
emacs -nw foo.go
emacs -Q -nw foo.go
emacs foo.go
emacs -Q foo.go
```

All 4 variants open Emacs and visit a file `foo.go`.

> arguments are ignored.

They aren't. The `"$@"` passes all the arguments to `Emacs`.

> Why using bash anyway when default shell in macOS now is zsh, not sure if that's relevant.

Completely irrelevant.

### solarized-dark not working anymore

I am pretty sure it's configuration issue. Might be also related to native compilation 🤷 

### init is partially ignored

I am pretty sure it's configuration issue. Might be also related to native compilation 🤷 

Try starting emacs with `--debug-init` option. Maybe you have errors you didn't notice. Also make sure to checkout your `*Messages*` buffer.

If init file is indeed partially ignored, it's better to report this to Emacs bug tracker. Emacs+ is just a formula for GNU Emacs 😄 But in case it's because of native compilation - double check that you are using it correctly.

### Spotlight search

See https://github.com/d12frosted/homebrew-emacs-plus/issues/618#issuecomment-1735892104, which contains multiple links to the same issue. It's a known thing. Status? Someone interested in fixing it needs to step in. 

### Just a few random comments

You don't need `--with-dbus` unless you really understand why you use it.

In addition, `--with-native-comp` might meddle with your configs. You need to properly setup it for your configs to work.

P.S. That reminds me that I need to enable AOT for native comp. Doesn't fix issues with configs, but at least makes the life easier.
